sidecar/internal/csiaddonsnode/csiaddonsnode.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csiaddonsnode
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	podNameEnvKey      = "POD_NAME"
+	podNamespaceEnvKey = "POD_NAMESPACE"
+	podUIDEnvKey       = "POD_UID"
+	podIPEnvKey        = "POD_IP"
+)
+
+// Deploy creates CSIAddonsNode custom resource withh all required information.
+func Deploy(config *rest.Config, driverName, nodeID, endpoint string) error {
+	object, err := getCSIAddonsNode(driverName, endpoint, nodeID)
+	if err != nil {
+		return err
+	}
+
+	err = csiaddonsv1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return errors.Wrapf(err, "failed to add scheme")
```

Comment:
Just use the new `fmt.Errorf("... : %w", err)` wrapping. No need to import `github.com/pkg/errors`